src/components/ToggleButton.js

```diff
@@ -0,0 +1,36 @@
+import React, {PropTypes} from 'react'
+import styles from './ToggleButton.css'
+import CSSModules from 'react-css-modules'
+import {changeTokenType} from '../actions/actions.js'
+import getID from '../utils/getID'
+
+class ToggleButton extends React.Component {
+  componentWillMount () {
+    this.state = {isChecked: false}
+    this.onChange = this.onChange.bind(this)
+    this.toggleId = getID()
+  }
+  onChange (e) {
+    let nextVal = !this.state.isChecked
+    this.setState({
+      isChecked: nextVal
+    })
+    // we need to inverse the nextVal because checked means isPreview === false
+    this.props.changeHandler(!nextVal)
+    changeTokenType(!nextVal)
+  }
+  render () {
+    return (
+      <div>
+        <div styleName='onoffswitch'>
+          <input type='checkbox' name='onoffswitch' styleName='onoffswitch-checkbox' id={this.toggleId} checked={this.state.isChecked} onChange={this.onChange}/>
```

Comment:
I'd get rid of `id` and `for` altogether. If you need label (we don't have a label text, so I don't see a point) then you can just nest: `<label><input type="checkbox /></label>` - it's completely valid HTML and binds label with an input.